Source/DafnyStandardLibraries/src/Std/Collections/SeqExt.dfy

```diff
@@ -0,0 +1,76 @@
+/*******************************************************************************
+ *  SeqExt - Extensions to Std.Collections.Seq
+ *  
+ *  This module adds MaxBy and MinBy functions to the Seq functionality
+ *  without redefining the entire Seq module (which would cause duplicate errors).
+ *
+ *  SPDX-License-Identifier: MIT
+ *******************************************************************************/
+
+module Std.Collections.SeqExt {
+  import opened Wrappers
+
+  /**********************************************************
+   *
+   *  Finding Maximum and Minimum Elements
+   *
+   ***********************************************************/
+
+/// Find the maximum element in a sequence using a comparator function
+///
+/// The comparator function should return:
+///   -1 if a < b (first argument is less than second)
+///    0 if a == b (equal)
+///    1 if a > b (first argument is greater than second)
+///
+/// Example usage with Duration:
+///   var maxDuration := MaxBy(durations, Duration.Compare)
+  function MaxBy<T>(s: seq<T>, comparator: (T, T) -> int): T
```

Comment:
In all of these it would be simpler and more consistent with other standard libraries to use a binary predicate instead of a comparator that returns a `-1`, `0`, or `1`. Either a `lessThanOrEq: (T, T) -> bool` as in [MergeSortBy](https://github.com/dafny-lang/dafny/blob/master/Source/DafnyStandardLibraries/src/Std/Collections/Seq.dfy#L989) or a `less: (T, T) -> bool` as in [`BinarySearch`](https://github.com/dafny-lang/dafny/blob/master/Source/DafnyStandardLibraries/src/Std/Collections/Array.dfy#L15)

That's easy enough to use from the Duration module by just using `Duration.Less` instead of `Duration.Compare`